frontend/src/components/UserUpdateRoles.vue

```diff
@@ -1,210 +1,384 @@
+                      </v-row>
+                    </v-col>
+                    <!-- effective roles -->
+                    <v-col class="col-4" no-gutters>
+                      <v-row>
+                        <label>
+                          Effective Roles
+                          <!-- tooltip for effective roles -->
+                          <v-tooltip right>
+                            <template v-slot:activator="{ on }">
+                              <v-icon v-on="on" small>mdi-help-circle</v-icon>
+                            </template>
+                            <span>
+                              Effective roles represent all roles assigned to a
+                              user for this client.
+                              <br />This may include roles provided by group
+                              membership which cannot be directly removed.
+                            </span>
+                          </v-tooltip>
+                        </label>
+                      </v-row>
+                      <!-- effective roles -->
+                      <v-row style="flex-direction: column">
+                        <v-col>
+                          <v-checkbox
+                            class="roles-checkbox"
+                            style="word-break:break-all"
+                            hide-details="auto"
+                            disabled
+                            readonly
+                            v-for="role in effectiveClientRoles"
+                            v-model="effectiveClientRoles"
+                            :value="role"
+                            :key="role.name"
+                          >
+                            <!-- effective role tooltip -->
+                            <span slot="label" class="tooltip">
+                              {{ role.name }}
+                              <span
+                                v-show="role.description"
+                                class="tooltiptext"
+                              >
+                                {{ role.description }}
+                              </span>
+                            </span>
+                          </v-checkbox>
+                        </v-col>
+                      </v-row>
+                    </v-col>
+                  </v-row>
+
+                  <!-- button to save new role assignments -->
+                  <div v-if="selectedClient">
+                    <v-btn
+                      v-if="hasRoleForManageUserRoles"
+                      id="save-user-roles"
+                      class="primary"
+                      medium
+                      v-on:click="updateUserClientRoles()"
+                      >Save User Roles</v-btn
+                    >
+                  </div>
+                </div>
+              </div>
+            </v-card>
+          </v-dialog>
+        </v-toolbar>
+      </template>
+      <template #item.lastLogin="{ item }">
```

Comment:
Can we swap the order of the lastLogin <template> block with the roleArray <template> block? 
It doesn't actually make a difference, but then their order in the file matches the order of the columns in the table itself.